.github/workflows/docker-build-test.yaml

```diff
@@ -307,17 +301,116 @@ jobs:
+
+            # Now do the check whether the target branch exists, outside the if-else
+            if git ls-remote --heads origin "aptos-release-v${target_branch}" | grep -q "refs/heads/${target_branch}"; then
+                echo "Target branch is $target_branch"
+            else
+                echo "Branch $target_branch does not exist"
+            fi
+          else
+            echo "Not merging to main or a release branch, using the main branch for compat test" >&2
+            target_branch="main"
+          fi
+          echo "TARGET_BRANCH=${target_branch}" >> $GITHUB_OUTPUT
+          echo "Determined TARGET_BRANCH: ${target_branch}" >&2
+
+      # actions/get-latest-docker-image-tag requires docker utilities and having authenticated to internal docker image registries
+      - uses: aptos-labs/aptos-core/.github/actions/docker-setup@main
+        id: docker-setup
+        with:
+          GCP_WORKLOAD_IDENTITY_PROVIDER: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+          GCP_SERVICE_ACCOUNT_EMAIL: ${{ secrets.GCP_SERVICE_ACCOUNT_EMAIL }}
+          EXPORT_GCP_PROJECT_VARIABLES: "false"
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DOCKER_ARTIFACT_REPO: ${{ secrets.AWS_DOCKER_ARTIFACT_REPO }}
+          GIT_CREDENTIALS: ${{ secrets.GIT_CREDENTIALS }}
+
+      - name: Get Docker Image Tag
+        uses: ./.github/actions/get-latest-docker-image-tag
+        id: get-docker-image-tag
+        with:
+          branch: ${{ steps.determine-the-target-branch.outputs.TARGET_BRANCH }}
+          variants: "failpoints performance"
+
+      - name: Log TARGET_BRANCH and IMAGE_TAG
+        run: |
+          echo "TARGET_BRANCH: ${{ steps.determine-the-target-branch.outputs.TARGET_BRANCH }}"
+          echo "IMAGE_TAG: ${{ steps.get-docker-image-tag.outputs.IMAGE_TAG }}"
+
+      - name: Set IMAGE_TAG as job output
+        run: echo "IMAGE_TAG=${{ steps.get-docker-image-tag.outputs.IMAGE_TAG }}" >> $GITHUB_OUTPUT
+
+    outputs:
+      IMAGE_TAG: ${{ steps.get-docker-image-tag.outputs.IMAGE_TAG }}
+  
+  # Run e2e compat test against testnet branch. This is a PR required job.
+  forge-compat-test:
+    needs: 
+      - fetch-last-released-docker-image-tag
+      - permission-check
+      - determine-docker-build-metadata
+      - rust-images
+      - rust-images-failpoints
+      - rust-images-performance
+      # - rust-images-consensus-only-perf-test
+      - file_change_determinator
     uses: aptos-labs/aptos-core/.github/workflows/workflow-run-forge.yaml@main
     secrets: inherit
     with:
       GIT_SHA: ${{ needs.determine-docker-build-metadata.outputs.gitSha }}
       FORGE_TEST_SUITE: compat
-      IMAGE_TAG: 25a081116546670e62ca927ba90478de78557056 #aptos-node-v1.20
+      IMAGE_TAG: ${{ needs.fetch-last-released-docker-image-tag.outputs.IMAGE_TAG }}
       FORGE_RUNNER_DURATION_SECS: 300
-      COMMENT_HEADER: forge-compat
```

Comment:
This header is necessary for the step that posts comments to the PR. We could derive it from the test suite though. It just needs to be unique per-PR-per-workflow, and we likely don't run the same test suite multiple times in a workflow: https://github.com/marocchino/sticky-pull-request-comment#keep-more-than-one-comment